docs/reference/ingest/processors/geo-grid.asciidoc

```diff
@@ -0,0 +1,250 @@
+[role="xpack"]
+[[ingest-geo-grid-processor]]
+=== Geo-grid processor
+++++
+<titleabbrev>Geo-grid</titleabbrev>
+++++
+
+Converts geo-grid definitions of grid tiles or cells to regular bounding boxes or polygons which describe their shape.
+This is useful if there is a need to interact with the tile shapes as spatially indexable fields.
+For example the `geotile` field value `"4/8/3"` could be indexed as a string field, but that would not enable
+any spatial operations on it.
+Instead we should convert it to the value
+`"POLYGON ((0.0 40.979898069620134, 22.5 40.979898069620134, 22.5 55.77657301866769, 0.0 55.77657301866769, 0.0 40.979898069620134))"`,
+which can be indexed as a <<geo-shape,`geo_shape`>> field.
+
+[[geo-grid-processor-options]]
+.geo_grid processor options
+[options="header"]
+|======
+| Name                        | Required  | Default   | Description
+| `field`                     | yes       | -         | The field to interpret as a geo-tile. The field format is determined by the `tile_type`.
+| `tile_type`                 | yes       | -         | Three tile formats are understood: `geohash`, `geotile` and `geohex`.
+| `target_field`              | no        | `field`   | The field to assign the polygon shape to, by default `field` is updated in-place.
+| `parent_field`              | no        | -         | If specified and a parent tile exists, save that tile address to this field.
+| `children_field`            | no        | -         | If specified and children tiles exist, save those tile addresses to this field as an array of strings.
+| `non_children_field`        | no        | -         | If specified and intersecting non-child tiles exist, save their addresses to this field as an array of strings.
+| `precision_field`           | no        | -         | If specified, save the tile precision (zoom) as an integer to this field.
+| `ignore_missing`            | no        | -         | If `true` and `field` does not exist, the processor quietly exits without modifying the document.
+| `target_format`             | no        | "GEOJSON" | Which format to save the generated polygon in. Either `WKT` or `GeoJSON`.
```

Comment:
I'm assuming this is case-insensitive, but maybe it would be good to be consistent between "GEOJSON" and `GeoJSON`?